main/util/src/mill/util/CoursierSupport.scala

```diff
@@ -106,10 +106,10 @@ trait CoursierSupport {
         case Right(res) =>
           Result.Success(
             Agg.from(
-              res.files
-                .map(os.Path(_))
-                .filter(resolveFilter)
-                .map(PathRef(_, quick = true))
+              res.fullDetailedArtifacts
```

Comment:
This is the root of the change: Not only return the path, but also what dependency.

Drawbacks:
- Will this break plugins. I'm not quite sure how backwards compatible these methods need to be?
- `.map(_._2)` in many locations, making the code more mysterious. Better use a `case class`. Other ideas?

Generally: Wondering if this is the right approach to return more info about the resolved dependencies. Or if I missed a obvious other way?

Reply:
I think in the past we would add a new task that returned the richer data type (e.g. a `case class` or something), and the old task would just call the new task and extract the information it needs. Then whoever needs the richer data type can call the new task

Reply:
Will do that. The only concern is that there are many 'flavors' of resolveDependencies. (CoursierSupport, Lib, JVM)
Anyway, can be cleared once the PR isn't draft anymore.